test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs

```diff
@@ -281,5 +282,107 @@ public async Task ThrowsIfUsingWrongExpiresOrRefresh()
+            }
+        }
+
+        [RavenTheory(RavenTestCategory.ExpirationRefresh)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        public async Task RefreshWithMaxItemsToProcessConfiguredShouldWork(Options options)
+        {
+            using (var store = GetDocumentStore(options))
+            {
+                // Insert documents with refresh before activating the refresh
+                var refresh = SystemTime.UtcNow.AddMinutes(5);
+                for (int i = 0; i < 10; i++)
+                {
+                    using (var session = store.OpenAsyncSession())
+                    {
+                        var company = new Company { Name = "Company Name", Id = $"companies/{i}" };
+                        await session.StoreAsync(company);
+                        var metadata = session.Advanced.GetMetadataFor(company);
+                        metadata[Constants.Documents.Metadata.Refresh] = refresh.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                        await session.SaveChangesAsync();
+                    }
+                }
+
+                var config = new RefreshConfiguration()
+                {
+                    Disabled = false,
+                    RefreshFrequencyInSec = TimeSpan.FromMinutes(10).Ticks,
+                    RefreshMaxItemsToProcess = 9
+                };
+
+                await RefreshHelper.SetupExpiration(store, Server.ServerStore, config);
+
+                var database = await GetDatabase(store.Database);
+                database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+
+                DateTime time = SystemTime.UtcNow.AddMinutes(10);
+                using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (context.OpenWriteTransaction())
+                {
+                    var refreshOptions = new ExpirationStorage.ExpiredDocumentsOptions(context, time, true, 10, 10);
+                    var toRefresh = database.DocumentsStorage.ExpirationStorage.GetDocumentsToRefresh(refreshOptions, out _, out var totalCount, CancellationToken.None);
+                    Assert.Equal(10, totalCount);
+                }
+
+                var expiredDocumentsCleaner = database.ExpiredDocumentsCleaner;
+                await expiredDocumentsCleaner.RefreshDocs();
+
+                using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (context.OpenWriteTransaction())
+                {
+                    var refreshOptions = new ExpirationStorage.ExpiredDocumentsOptions(context, time, true, 10, 10);
+                    var toRefresh = database.DocumentsStorage.ExpirationStorage.GetDocumentsToRefresh(refreshOptions, out _, out var totalCount, CancellationToken.None);
+                    Assert.Equal(1, totalCount);
+                }
+            }
+        }
     }
```

Comment:
We may add a test for storing and "expiring" 10 docs, with settings: batchSize = 5, maxItemsToProcess=9, which ensures we removed only 9 docs